static/js/shared/data_fetcher.ts

```diff
@@ -420,16 +449,39 @@ function fetchStatsData(
       }
       // Build the dates collection, get the union of available dates for all data
       for (const place in allResp[i].data) {
+        if (!(place in numStatVarsPerPlace)) {
+          numStatVarsPerPlace[place] = 0;
+        }
         if (!allResp[i].data[place]) continue;
         const timeSeries = allResp[i].data[place];
+        if (Object.keys(timeSeries.data).length > 0) {
+          numStatVarsPerPlace[place] = numStatVarsPerPlace[place] + 1;
+        }
         result.sources.add(timeSeries.provenanceDomain);
         for (const date in timeSeries.data) {
-          dates[date] = true;
+          if (date in dates) {
+            dates[date] = dates[date] + 1;
+          } else {
+            dates[date] = 1;
+          }
         }
       }
     }
     result.dates = Object.keys(dates);
     result.dates.sort();
+    result.dateToSelect = result.dates[result.dates.length - 1];
```

Comment:
Move this to line 485, so it's clear that this is the default case when we can't find a better date.